tools/codegen/dest/ts_lowering.py

```diff
@@ -0,0 +1,104 @@
+from typing import List, Optional, Union, Tuple
+import itertools
+from typing_extensions import Literal
+from dataclasses import dataclass
+from enum import Enum
+import textwrap
+from tools.codegen import local
+from tools.codegen.context import method_with_native_function, native_function_manager
+from tools.codegen.utils import Target, mapMaybe
+from tools.codegen.model import (BaseType, OptionalType, DispatchKey, NativeFunction,
+                                 NativeFunctionsGroup, SchemaKind, FunctionSchema,
+                                 TensorOptionsArguments, ListType,
+                                 DeviceCheckType, Argument, assert_never,
+                                 is_cuda_dispatch_key, BackendIndex,
+                                 gets_generated_out_inplace_wrapper, OperatorName,
+                                 SelfArgument, Arguments)
+from tools.codegen.api.types import (BaseTy, BaseCppType, BaseCType, OptionalCType,
+                                     Binding, ConstRefCType, NamedCType,
+                                     CppSignature, CppSignatureGroup,
+                                     Expr, MutRefCType, kernel_signature,
+                                     DispatcherSignature, ListCType)
+import tools.codegen.api.meta as meta
+import tools.codegen.api.cpp as cpp
+import tools.codegen.api.structured as structured
+from tools.codegen.api.translate import translate
+from tools.codegen.selective_build.selector import SelectiveBuilder
+from .lazy_ir import ir_node_name, valueT, valueListT, update_schema_for_lazy_ir, isValueType
+
+def separate_args_kwargs_types(func: FunctionSchema) -> Tuple[List[NamedCType], List[NamedCType], List[NamedCType], List[NamedCType], List[NamedCType]]:
+    """
+    Since there are significant differences in how Values and 'everything else' are handled in the IR node class,
+    it's useful to have a way to get a list of just the values, just the scalars, or all of them.
+    """
+    args = func.arguments
+    types = [arg.type for arg in func.schema_order_arguments()]
+    positional_values = [t.type for t in args.flat_positional if isValueType(t.type)]
+    positional_scalars = [t.type for t in args.flat_positional if not isValueType(t.type)]
+    kw_values = [t.type for t in args.flat_kwarg_only if isValueType(t.type)]
+    kw_scalars = [t.type for t in args.flat_kwarg_only if not isValueType(t.type)]
+    return types, positional_values, positional_scalars, kw_values, kw_scalars
+
+@dataclass(frozen=True)
+class TsLowering:
+    backend_index: BackendIndex
+
+    # Names of operators we want to codegen for, a subset of backend_index
+    codegen: List[OperatorName]
```

Comment:
nit: instead of storing the list of codegen'd op names here, you could probably use it directly in `gen_backend_stubs.py` to filter which NativeFunctions get passed into this class. So the invariant is `TsLowering(f)` is only called if f is meant to be codegen'd